Sources/PT.PM.Matching/PatternRoot.cs

```diff
@@ -120,7 +120,7 @@ private static void TraverseChildren(PatternUst patternUst, Ust ust, MatchContex
                 }
 
                 var match = new MatchResult(
-                    ust is RootUst rootUst ? rootUst : ust.Root, context.PatternUst, context.Locations);
+                    ust.RootOrThis, context.PatternUst, context.Locations);
```

Comment:
i think better way is use `ust.Root` property, and just override it in RootUst to `get => return this;`

Reply:
`RootUst` can have `Root` property itself in mixed-language files like JavaScript in PHP. In this case, JavaScript `RootUst` have a not null `Root` property that points to `RootUst` of PHP.

Reply:
so RootUst can include private field root, and property can look like get => root ?? this.